src/node/MosaicProxyApp.cc

```diff
@@ -171,8 +170,12 @@ void MosaicProxyApp::receivePacket(omnetpp::cMessage *msg) {
   EV << "MosaicUDP received packet: ";
 
   auto udp_packet = inet::check_and_cast<inet::Packet *>(msg);
-  auto cPacket = udp_packet->popAtBack<inet::cPacketChunk>().get()->getPacket();
+  auto cPacket =
+      udp_packet->popAtBack<inet::cPacketChunk>(udp_packet->getDataLength())
```

Comment:
How is omnet interpreting packet length in general? bits or bytes? Are we sure it's handled everywhere correctly?